.github/scripts/determine-slack-payload-for-package.ts

```diff
@@ -0,0 +1,215 @@
+  );
+  return content.trim();
+}
+
+/**
+ * Parses the output from a previous `module-lint` run into parts that can be
+ * used to construct a Slack message.
+ *
+ * @param moduleLintOutput - The output from a previous `module-lint` run.
+ * @returns The pieces of the `module-lint` output.
+ */
+function parseModuleLintOutput(
+  moduleLintOutput: string,
+): ParsedModuleLintOutput {
+  const moduleLintOutputSections = moduleLintOutput.split('\n\n');
+
+  const summarySection =
+    moduleLintOutputSections[moduleLintOutputSections.length - 1];
+  if (summarySection === undefined) {
+    throw new Error("Couldn't parse module-lint report output");
+  }
+
+  const [reportSummaryLine, elapsedTimeLine] = summarySection.split('\n');
+  if (reportSummaryLine === undefined || elapsedTimeLine === undefined) {
+    throw new Error("Couldn't parse module-lint report output");
+  }
+
+  const reportSummary = reportSummaryLine.replace(/^Results:[ ]+/u, '');
+  const reportSummaryMatch = reportSummary.match(
+    /(\d+) passed, (\d+) failed, (\d+) errored, (\d+) total/u,
+  );
+  if (!reportSummaryMatch) {
+    throw new Error("Couldn't parse module-lint report output");
+  }
+
+  const passed = Number(reportSummaryMatch[1]);
+  const failed = Number(reportSummaryMatch[2]);
+  const errored = Number(reportSummaryMatch[3]);
+  const total = Number(reportSummaryMatch[4]);
+  const durationWithUnit = elapsedTimeLine.replace(/^Elapsed time:[ ]+/u, '');
+
+  const percentage = Math.round((passed / total) * 1000) / 10;
+
+  return { passed, failed, errored, total, durationWithUnit, percentage };
+}
+
+/**
+ * Constructs the payload that will be used to post a message in Slack
+ * containing information about the `module-lint` run for the package in
+ * question.
+ *
+ * @param inputs - The inputs to this script.
+ * @param moduleLintOutput - The output from a previous `module-lint` run.
+ * @returns The Slack payload.
+ */
+function constructSlackPayload(inputs: Inputs, moduleLintOutput: string) {
+  const { passed, total, percentage } = parseModuleLintOutput(moduleLintOutput);
+
+  const text = `Report for MetaMask/${inputs.projectName}`;
+
+  const blocks = [
+    {
+      type: 'header',
+      text: {
+        type: 'plain_text',
+        text: `MetaMask/${inputs.projectName}`,
+      },
+    },
+    {
+      type: 'rich_text',
+      elements: [
+        {
+          type: 'rich_text_section',
+          elements: [
+            {
+              type: 'emoji',
+              name: passed === total ? 'white_check_mark' : 'x',
+            },
+            {
+              type: 'text',
+              text: ` ${passed}/${total} rules passed (${percentage}% alignment with template).\n\n`,
+              style: {
+                bold: true,
+              },
+            },
+          ],
+        },
+        {
+          type: 'rich_text_preformatted',
+          elements: [
+            {
+              type: 'text',
+              text: moduleLintOutput,
```

Comment:
@kanthesha Currently this will dump the whole output of `module-lint` into the Slack message (e.g. https://consensys.slack.com/archives/C06S195ANA0/p1720587221996819?thread_ts=1720587196.429879&cid=C06S195ANA0). I know we discussed moving this out of the message and linking to the run instead, but there doesn't seem to be a way to do this. [You can generate a link to the entire run] (https://stackoverflow.com/a/70566764/260771), but you can't generate a link to a _job_ within a run. This has been a community-requested feature for a while and people have come up with [hacky solutions](https://stackoverflow.com/a/76681922/260771), but there doesn't seem to be an official way to do this. So, thoughts on just leaving this here for now?

Reply:
Okay, I've removed the output from the Slack messages. You can see the output here: https://consensys.slack.com/archives/C06S195ANA0/p1721751612763019